Jamulus.pro

```diff
@@ -95,20 +95,39 @@ win32 {
     contains(CONFIG, "jackonwindows") {
         message(Using Jack instead of ASIO.)
 
-        !exists("C:/Program Files (x86)/Jack/includes/jack/jack.h") {
-            message(Warning: jack.h was not found at the usual place, maybe jack is not installed)
+        contains(QT_ARCH, "x86") {
+		    !exists("C:/Program Files (x86)/JACK2/includes/jack/jack.h") {
+                message(Warning: jack.h was not found at the usual place, maybe jack2 32-bit is not installed)
+            }
+
+            HEADERS -= windows/sound.h
```

Comment:
Get rid of this duplication.

Reply:
To get rid of this duplication, I will also have to use a `lib-name` variable. This variable should return :

- libjack64.lib for x86_64,
- libjack.lib for x86.

So ... Will do when it's ok in my head.

Reply:
`        contains(QT_ARCH, "x86") {
        	ProgramFilesDir = read_registry(HKLM, ProgramFilesDir[, 32])
        	libjack-name = libjack.lib
		    !exists("$${ProgramFilesDir}/JACK2/include/jack/jack.h") {
                message(Warning: "jack.h was not found at the usual place, maybe JACK2 is not installed")
            }

            HEADERS -= windows/sound.h
            SOURCES -= windows/sound.cpp
            HEADERS += linux/sound.h
            SOURCES += linux/sound.cpp
            DEFINES += WITH_JACK
            DEFINES += JACK_REPLACES_ASIO
            DEFINES += _STDINT_H # supposed to solve compilation error in systemdeps.h
            INCLUDEPATH += "$${ProgramFilesDir}/JACK2/include"
            LIBS += "$${ProgramFilesDir}/JACK2/lib/$${libjack-name}"
        }

        contains(QT_ARCH, "x86_64") {
		    ProgramFilesDir = read_registry(HKLM, ProgramFilesDir[, 64])
		    libjack-name = libjack64.lib
		    !exists("$${ProgramFilesDir}/JACK2/include/jack/jack.h") {
                message(Warning: "jack.h was not found at the usual place, maybe JACK2 64-bit is not installed")
            }

            HEADERS -= windows/sound.h
            SOURCES -= windows/sound.cpp
            HEADERS += linux/sound.h
            SOURCES += linux/sound.cpp
            DEFINES += WITH_JACK
            DEFINES += JACK_REPLACES_ASIO
            DEFINES += _STDINT_H # supposed to solve compilation error in systemdeps.h
            INCLUDEPATH += "$${ProgramFilesDir}/JACK2/include"
            LIBS += "$${ProgramFilesDir}/JACK2/lib/$${libjack-name}"
        }`

Reply:
c0a5110